src/App.jsx

```diff
@@ -14,45 +15,81 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [currentOrder, setCurrentOrder] = useState([...goodsFromServer]);
+  const [isOriginalOrder, setIsOriginalOrder] = useState(true);
+
+  const sortAlphabetically = () => {
+    const newOrder = [...currentOrder].sort();
+
+    setCurrentOrder(newOrder);
+    setIsOriginalOrder(false);
+  };
+
+  const sortByLength = () => {
+    const newOrder = [...currentOrder].sort((a, b) => a.length - b.length);
+
+    setCurrentOrder(newOrder);
+    setIsOriginalOrder(false);
+  };
+
+  const reverseOrder = () => {
+    const newOrder = [...currentOrder].reverse();
+
+    setCurrentOrder(newOrder);
+    setIsOriginalOrder(!isOriginalOrder);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetOrder = () => {
+    if (!isOriginalOrder) {
+      setCurrentOrder([...goodsFromServer]);
+      setIsOriginalOrder(true);
+    }
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${isOriginalOrder ? 'is-light' : ''}`}
+          onClick={sortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${isOriginalOrder ? 'is-light' : ''}`}
+          onClick={sortByLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${isOriginalOrder ? 'is-light' : ''}`}
+          onClick={reverseOrder}
+        >
+          Reverse
+        </button>
+
+        <button
+          type="button"
+          className={`button is-danger ${isOriginalOrder ? 'is-light' : ''}`}
+          onClick={resetOrder}
+        >
+          Reset
+        </button>
+      </div>
```

Comment:
you should only show this button if your list is already being sorted
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/145850632/3da2f89c-0325-4c0d-b19e-20db2c21c8c2)